### PR TITLE
Pac interface

### PR DESCRIPTION
**Pull request by robinbaeyens**
### Include PAC chemistry interfacing functions into the main gcm_toolkit. 

**PACInterface**

New class `PACInterface` that implements `Interface` and contains mainly functions to write PAC-specific input files, such as:

- `write_inputfile` (1D or pseudo-2D)
- `generate_lptfile`
- `generate_aptfiles`

**Thermosphere**

New function `utils.manipulations.m_extend_upward` that provides functionality for a low-pressure, parametrized thermosphere extension onto existing datasets. An isothermal extension is also possible. 

**Miscelaneous**

Small helper method `_extract_jet_speed` to average the equatorial jet speed into a single number.

A helper method `_p_to_alt` to convert pressures to altitudes, assuming a non-constant gravity. In the future, it could be a good idea to make this a more general manipulation rather than an interface-specific method, see [this discussion](https://github.com/exorad/gcm_toolkit/commit/cb2fc8abdae585295355af076db2c0e1806fd2a3#r86757516).

### CODE REVIEW

**Review by Kiefersv — COMMENTED** [review]
Hey Robin,

I had a look through it and it looks good so far! Have to admit I don't really know if all the PAC stuff is correct but it is well added to the gcm_toolkit :D

File: gcmt/utils/manipulations.py [hunk 1]
```diff
@@ -86,3 +86,121 @@ def m_add_meridional_overturning(dsi, v_data='V', var_key_out=None):
         dsi.update({var_key_out: psi})
 
     return psi
+
+
+def m_extend_upward(temp_array, p_low, method=None, n_p=20, T_therm=None,
+                    p_therm_high=None):
```
[the remark this answers is not in the document]
Comment (AaronDavidSchneider) [reply gcmt/utils/manipulations.py:92]: I agree with @Kiefersv!

**Comment by AaronDavidSchneider** [comment]
Hey @robinbaeyens, I know its a littlebit annoying to do, but could you fix the merge conflicts? I can help, if you need help with that. In principle, the cleanest is if you try to do a git rebase onto the current version of the main branch and then force push again.

**Comment by robinbaeyens** [comment]
> Hey @robinbaeyens, I know its a littlebit annoying to do, but could you fix the merge conflicts? I can help, if you need help with that. In principle, the cleanest is if you try to do a git rebase onto the current version of the main branch and then force push again.

Hey @AaronDavidSchneider of course I'll fix the conflicts. I'll try not to break stuff and let you know if it doesn't work out!

**Comment by robinbaeyens** [comment]
The merge conflicts should now be fixed. It was a huge pain to rebase/merge because I had branched off before the renaming of most files to gcm_toolkit happened, which didn't help. 
I added testing for the `extend_upward()` method, but not for the PAC interface because that's not worth the effort. 

@Kiefersv  if you can check and approve, we can merge the branch :)

File: gcm_toolkit/tests/test_manipulations.py [hunk 2]
```diff
@@ -164,3 +164,27 @@ def test_horizontal_overturning(all_nc_testdata):
...
+    assert len(T_iso.Z) == len(dsi.Z) + 20
+    assert min(T_iso.Z) == 1e-8
+
+    T_the = tools.extend_upward(1e-8, method='thermosphere',
+                                  T_therm=10000, p_therm_high=1e-6)
+
+    assert len(T_the.Z) == len(dsi.Z) + 20
+    assert min(T_the.Z) == 1e-8
+    assert (T_the.sel(Z=slice(1e-6,1e-8)) < 10000).all()
+    assert (T_the.sel(Z=slice(1e-6,1e-8)) >= T_iso.sel(Z=slice(1e-6,1e-8))).all()
```
[the remark this answers is not in the document]
Comment (Kiefersv) [reply gcm_toolkit/tests/test_manipulations.py:190]: That would be a nice improvement but I'm also ok with this version.

My bigger problem is that, according to codecov, your testing hits virtually no line in gcm_toolkit/utils/interface.py. Could you add testing to the PACInterface class as well?

**Review by Kiefersv — COMMENTED** [review]
The changes to look good. I assume that the PAC routines perform as expected on your end, I was not able to check that of course :D

As mentioned in the comment, there are no tests yet for the PAC class. Could you still add them?

Edit: Just saw your replay about the testing of the PAC routines. I see the point that they are not the most important thing to test. Would it be easy to just add a quick call of the PACinterface that tests if it at least it is set up correctly?

File: gcm_toolkit/tests/test_manipulations.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (robinbaeyens) [reply gcm_toolkit/tests/test_manipulations.py:190]: Thanks for looking over this. 

Yeah, I realise that the PAC interface is not tested, but this was a conscious decision. Most of the routines do similar stuff, namely writing out files. I spent a bit of time figuring out how to test for such functionality, but it involves setting up some mock I/O stuff that I frankly don't think is worth the time and effort. Especially since I'll be one of the only people using it. 

In short: I don't want to write tests for it, and I hope that's okay :) Let me know what you think.

[the remark this answers is not in the document]
Comment (AaronDavidSchneider) [reply gcm_toolkit/tests/test_manipulations.py:190]: No, I agree! That's perfectly fine @robinbaeyens! I was only referring to the tests of extend_upwards 😊

**Comment by robinbaeyens** [comment]
Basic testing for the PAC interface is also done now! @Kiefersv can you have another look please? :)

**Review by Kiefersv — APPROVED** [review]
Thanks for your efforts, it looks good to me!

**Comment by AaronDavidSchneider** [comment]
Thank you @robinbaeyens!
